### PR TITLE
remove abc from lightning module

### LINKED ISSUE #9499: Remove ABC from Lightning Module
## Proposed refactoring or deprecation

No longer extend ABC in LightningModule:
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/core/lightning.py#L57

### Motivation

According to @carmocca in https://docs.google.com/document/d/1xHU7-iQSpp9KJTjI3As2EM0mfNHHr37WZYpDpwLkivA/edit?disco=AAAANOyYJFs this is legacy and can be removed. Originally added in https://github.com/PyTorchLightning/pytorch-lightning/pull/644 but we are no longer using the `@abstractmethod` decorator.

### PR DESCRIPTION
**Pull request by PuneetDabral**
## What does this PR do?

Follow up to #8752 
Fixes #9499

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]
we can make this change. however, as I commented in  #8752 the class remains abstract as long as we raise NotImplementedError in the dataloader methods.

**Review by tchaton — APPROVED** [review]
LGMT !

**Comment by ananthsub** [comment]
@PuneetDabral thanks for working on this! Could you update the CHANGELOG too?

**Review by SeanNaren — APPROVED** [review]

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -54,7 +54,6 @@
 
 
 class LightningModule(
-    ABC,
```
Comment (carmocca) [inline pytorch_lightning/core/lightning.py:57]: You'll need to remove the import too:

`pytorch_lightning/core/lightning.py:22: [F401] 'abc.ABC' imported but unused`

Comment (PuneetDabral) [reply pytorch_lightning/core/lightning.py:57]: it is my first time sir i am totally confuse please help me sir ....

**Review by carmocca — APPROVED** [review]

**Comment by PuneetDabral** [comment]
now my contribution is done....

**Comment by carmocca** [comment]
> sir now my contribution is done? ...

Yes, it is! Waiting for the tests to finish.

**Comment by PuneetDabral** [comment]
ok sir please tell me ...i eagerly waiting

**Comment by awaelchli** [comment]
can you merge master into this branch and push again to trigger a rerun on the last check here? 

```bash
git checkout master
git pull
git checkout remove/abc
git merge master
git push
```

**Review by akihironitta — COMMENTED** [review]
Once the remaining test gets green, LGTM.

**Comment by PuneetDabral** [comment]
ok i will commit again after  2-3 hour ....

**Comment by PuneetDabral** [comment]
git push origin remove/abc
Everything up-to-date

sir this will appear....

**Comment by awaelchli** [comment]
ah yes sir you need to pull the new changes from the master branch first and then merge it. see my instruction from the previous message.

**Comment by PuneetDabral** [comment]
sir it was show all the things are up to data noting to commit

**Comment by PuneetDabral** [comment]
so sir what i can do?

**Comment by PuneetDabral** [comment]
it was again fail sir.. i am totally demotivated
